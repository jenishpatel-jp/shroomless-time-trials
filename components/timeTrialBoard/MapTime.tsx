import { View, Text, StyleSheet } from 'react-native'

interface MapTimeProp {
  time: string;
}

const MapTime: React.FC<MapTimeProp> = ( { time } ) => {

  return (
    <View style={styles.container}> 
        <Text style={styles.time} >{time}</Text>
    </View>

  )
}

export default MapTime

const styles = StyleSheet.create({
  container: {
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: 'black',
      color: 'black',
      width: 240,
      height: '100%',
      borderRadius: 10,
  },
  time: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
  }

})