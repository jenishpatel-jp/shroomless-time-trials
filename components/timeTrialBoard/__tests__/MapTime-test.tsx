import { render, screen } from "@testing-library/react-native";
import MapTime from "../MapTime";

describe("MapTime", () => {
    
    test("renders a time", () => {
        const times = ["00:00.000", "01:00:23", "3:43.123"];

        times.forEach((time) => {
            const { unmount } = render(<MapTime time={time} />);
            expect(screen.getByText(time)).toBeOnTheScreen();
            unmount();
        });
    });

    test("renders empty time string", () => {
        render(<MapTime time={""} />);
        expect(screen.getByText("")).toBeOnTheScreen();
    });


    test("matches snapshot", () => {
        const tree = render(<MapTime time="00:00.000" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("time has correct style", () => {
        render(<MapTime time="01:23.456" />);
        expect(screen.getByText("01:23.456")).toHaveStyle({
            fontSize: 20,
            fontWeight: "bold",
            color: "#009FE5",
        }); 
    });
})