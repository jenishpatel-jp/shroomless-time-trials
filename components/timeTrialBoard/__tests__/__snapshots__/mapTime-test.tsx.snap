// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapTime matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 10,
      "color": "black",
      "height": "100%",
      "justifyContent": "center",
      "width": 240,
    }
  }
>
  <Text
    style={
      {
        "color": "#009FE5",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    00:00.000
  </Text>
</View>
`;
